pkg/network/ebpf/c/protocols/kafka/kafka-parsing.h

```diff
@@ -1804,4 +1808,22 @@ static __always_inline void update_topic_name_size_telemetry(kafka_telemetry_t *
     __sync_fetch_and_add(&kafka_tel->topic_name_size_buckets[bucket_idx], 1);
 }
 
+// update_classified_api_version_hits_telemetry updates arrays keeping track of api versions hits in the telemetry
+static __always_inline void update_classified_api_version_hits_telemetry(kafka_telemetry_t *kafka_tel, s16 api_key, s16 api_version) {
+    __u8 bucket_idx = api_version - 1;
```

Comment:
there's a type mismatch here between s16 and _u8
if you assume the api_version is positive and smaller than 255, then let's use __u8 all the way